src/base.c

```diff
@@ -70,27 +78,29 @@ base_unmap(tsdn_t *tsdn, extent_hooks_t *extent_hooks, unsigned ind, void *addr,
 		if (extent_hooks->dalloc != NULL &&
 		    !extent_hooks->dalloc(extent_hooks, addr, size, true,
 		    ind)) {
-			goto label_done;
-		}
-		if (extent_hooks->decommit != NULL &&
+			/* Fall through */
```

Comment:
The logic was cascading through the possible methods for "free" the mappings, and stop if any of them succeeded. I'll refactor this part a bit.